lib/Lec6-BinaryHeap/binaryHeap.js

```diff
@@ -0,0 +1,41 @@
+let array_length;
+let heap_root = (input, i) => {
+    let left = 2 * i + 1;
+    let right = 2 * i + 2;
+    let maxValue = i;
+
+    if (left < array_length && input[left] > input[maxValue]) {
+        maxValue = left;
+    }
+    if (right < array_length && input[right] > input[maxValue]) {
+        maxValue = right;
+    }
+
```

Comment:
I think here should be more logic to chose item to swap with. You should swap "Parent Node" with "Child Node" (left or right) in dependence on result of "is left greater (or smaller) than right". In your case you should swap nodes when "Child Node" is greater than "Parent Node": `input[left/right] > input[maxValue]`. But you can have a case when child nodes (left and right) will be greater than parent. According to this logic you should firstly get the greatest node between "Left" and "Right" and only then you will be able to compare it with parent. It will prevent you from unexpected logic when you have, for instance: `left = 10; right = 7; max = 5`. In this case you should compare 7 and 10, choose 10 and then you will compare 10 and 5 and only now you can swap them. The result will be `max=10; left=5; right=7`